PWGCF/Flow/Tasks/flowAnalysisGF.cxx

```diff
@@ -0,0 +1,686 @@
+
+  void processData(myCollisions::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks)
+  {
+    registry.fill(HIST("hEventCount"), .5);
+
+    if (!collision.sel8())
+      return;
+
+    int Ntot = tracks.size();
+    if (Ntot < 1)
+      return;
+
+    const auto centrality = collision.centFT0C();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+
+    if (cfgUseAdditionalEventCut && !eventSelected(collision, tracks.size(), centrality))
+      return;
+    if (cfgFillQA)
+      FillEventQA(collision, tracks);
+    loadCorrections(bc.timestamp());
+    auto field = (cfgMagField == 99999) ? getMagneticField(bc.timestamp()) : cfgMagField;
+
+    processCollision<kReco>(collision, tracks, centrality, field);
+  }
+  PROCESS_SWITCH(flowAnalysisGF, processData, "Process analysis for non-derived data", true);
+
+  void processMCReco(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Mults, aod::CentFT0Cs>>::iterator const& collision, aod::BCsWithTimestamps const&, soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TrackSelection, aod::TracksDCA, aod::McTrackLabels>> const& tracks, aod::McParticles const&)
+  {
+    if (!collision.sel8())
+      return;
+    const auto centrality = collision.centFT0C();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    if (cfgUseAdditionalEventCut && !eventSelected(collision, tracks.size(), centrality))
+      return;
+    if (cfgFillQA)
+      FillEventQA(collision, tracks);
+    loadCorrections(bc.timestamp());
+    auto field = (cfgMagField == 99999) ? getMagneticField(bc.timestamp()) : cfgMagField;
+    processCollision<kReco>(collision, tracks, centrality, field);
+  }
+  PROCESS_SWITCH(flowAnalysisGF, processMCReco, "Process analysis for MC reconstructed events", false);
+
+  Filter mcCollFilter = nabs(aod::mccollision::posZ) < cfgVtxZ;
+  void processMCGen(soa::Filtered<aod::McCollisions>::iterator const& mcCollision, soa::SmallGroups<soa::Join<aod::McCollisionLabels, aod::Collisions, aod::CentFT0Cs>> const& collisions, aod::McParticles const& particles)
+  {
+    if (collisions.size() != 1)
+      return;
+    float centrality = -1;
+    for (auto& collision : collisions) {
+      centrality = collision.centFT0C();
+    }
+    processCollision<kGen>(mcCollision, particles, centrality, -999);
+  }
+  PROCESS_SWITCH(flowAnalysisGF, processMCGen, "Process analysis for MC generated events", false);
+
+  void processRun2(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Mults, aod::CentRun2V0Ms>>::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks)
+  {
+    if (!collision.sel7())
+      return;
+    const auto centrality = collision.centRun2V0M();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    loadCorrections(bc.timestamp());
+    auto field = (cfgMagField == 99999) ? getMagneticField(bc.timestamp()) : cfgMagField;
+    processCollision<kReco>(collision, tracks, centrality, field);
```

Comment:
Most probably this one can be also incorporated to the above mentioned unified reconstructed handling with the appropriate check of an existing column in the case of Run2 (as you check for the existence of the `McParticleId` in the case of reconstructed tracks for detecting MC detector level tracks)